08_education/digital_access.qmd

```diff
@@ -0,0 +1,957 @@
+        "other_digital",
+        "black_digital",
+        "hispanic_digital",
+        "white_digital",
+        "income_less_than_50k_digital",
+        "income_more_than_50k_digital"
+      )),
+      .fns = list(
+        lb = ~ .x - get(paste0("moe_", cur_column())),
+        ub = ~ .x + get(paste0("moe_", cur_column()))
+      ),
+      .names = "{.col}_{.fn}"
+    ))
+}
+
+
+lst(
+  county_overall,
+  place_overall,
+  county_raceeth,
+  place_raceeth,
+  county_income,
+  place_income
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = calculate_confidence_intervals(.y),
+      envir = .GlobalEnv
+    )
+  )
+```
+
+## Calculate the digital access metric
+
+```{r}
+#| label: calculate-digital-access-metric
+
+calculate_digital_access_metric <- function(data) {
+  data |>
+    mutate(across(
+      .cols = any_of(c(
+        "total_households",
+        "total_people_in_households",
+        "other",
+        "black",
+        "hispanic",
+        "white",
+        "income_less_than_50k",
+        "income_more_than_50k"
+      )),
+      .fns = list(
+        share_digital_access = ~ get(paste0(cur_column(), "_digital")) / .x,
+        share_digital_access_lb = ~ get(paste0(cur_column(), "_digital_lb")) / .x,
+        share_digital_access_ub = ~ get(paste0(cur_column(), "_digital_ub")) / .x
+      ),
+      .names = "{.fn}_{.col}"
+    )) |>
+    arrange(geoid, name, year) %>%
+    # replace instances with division by zero
+    mutate(across(everything(), ~ if_else(is.nan(.x) | is.infinite(.x), NA, .x)))
+}
+
+
+lst(
+  county_overall,
+  place_overall,
+  county_raceeth,
+  place_raceeth,
+  county_income,
+  place_income
+) %>%
+  map2(
+    .x = paste0(names(.), "_digital_access"),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = calculate_digital_access_metric(.y),
+      envir = .GlobalEnv
+    )
+  )
+
+
+map_dbl(county_overall_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_overall_digital_access, ~ sum(is.nan(.x)))
+map_dbl(county_raceeth_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_raceeth_digital_access, ~ sum(is.nan(.x)))
+map_dbl(county_income_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_income_digital_access, ~ sum(is.nan(.x)))
+```
+
+Inspect summary statistics and missigness for counties:
+
+```{r}
+map(
+  list(
+    county_overall_digital_access,
+    county_raceeth_digital_access,
+    county_income_digital_access
+  ),
+  skim
+)
+```
+
+and for places:
+
+```{r}
+map(
+  list(
+    place_overall_digital_access,
+    place_raceeth_digital_access,
+    place_income_digital_access
+  ),
+  skim
+)
+```
+
+# Create data quality flags
+
+We can use the width of the MOE to determine quality. If the value minus the
+lower bound is \<0.1, then the quality is strong (\_quality=1); if it is =\>.1
+and \<.2, the quality is marginal (\_quality=2), and if it is \>=.2, then the
+quality is weak (\_quality=3). These thresholds apply to both the 1 year and 5
```

Comment:
not urgent, but since we are going with 5 year, let's update this language back to just 5 year